rebasehelper/specfile.py

```diff
@@ -263,12 +263,14 @@ def __init__(self, path: str, sources_location: str = ''):
         self.extra_version: str = ''
         self.extra_version_separator: str = ''
         self.category: Optional[PackageCategory] = None
-        self.spc: rpm.spec = RpmHelper.get_rpm_spec(self.path)
-        self.header: RpmHeader = RpmHeader(self.spc.sourceHeader)
+        # initialize self.spc in self._update_data because some specs may require
+        # files from %sources when RPM parses the spec
+        self.spc: Optional[rpm.spec] = None
         self.spec_content: SpecContent = self._read_spec_content()
 
         # Load rpm information
         self._update_data()
+        self.header: RpmHeader = RpmHeader(self.spc.sourceHeader)
```

Comment:
This is rather redundant, since the attribute is already being assigned in `_update_data()`.
I think it should be made optional and initialized to `None`, just like `spc`.

Reply:
oki, this is the result after I rebased, I didn't look into details

Will change

Reply:
Unfortunately making those attributes optional triggers all sorts of typing errors, and there is no easy solution (I think). Better approach would be to move spec related code from `SpecFile._update_data()` to `RpmHelper.get_rpm_spec()`.

Reply:
@TomasTomecek I'll get to that if you don't mind, I have some more ideas for refactoring `_update_data()` and other stuff.